Products/GenericSetup/registry.py

```diff
@@ -644,6 +644,10 @@ def __init__(self):
     def getProfileInfo(self, profile_id, for_=None):
         """ See IProfileRegistry.
         """
+        if profile_id.startswith("profile-"):
+            profile_id = profile_id[len('profile-'):]
```

Comment:
I don't see `len(8)` being less fragile than `len('profile-'). 

Code like this has been here for years, see for example:
https://github.com/zopefoundation/Products.GenericSetup/blame/1.7.7/Products/GenericSetup/tool.py#L839
https://github.com/zopefoundation/Products.GenericSetup/blame/1.7.7/Products/GenericSetup/tool.py#L998